File: Homework/Ngan_Luu/905-sort-array-by-parity.py
#https://leetcode.com/problems/sort-array-by-parity 
class Solution:            
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        p=0
        for i in range(1,len(nums)):
            if nums[p]%2==0:
                p+=1
            elif nums[i]%2==0:
                nums[p],nums[i]=nums[i],nums[p]
                p+=1
        return nums

File: Homework/Son/W1/905-sort-array-by-parity.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
 var sortArrayByParity = function(nums) {
    let start = 0;
    let end = nums.length - 1;
    while (start !== end) {
        if (nums[start] % 2 === 0) start++;
        else if (nums[end] % 2 === 1) end--;
        if (nums[start] % 2 === 1 && nums[end] % 2 === 0) {
            let temp = nums[start]
            nums[start] = nums[end]
            nums[end] = temp
        }
    }
    return nums;
};

// https://leetcode.com/problems/sort-array-by-parity/